flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/dataview/ListView.java

```diff
@@ -156,17 +173,49 @@ public boolean equals(Object o) {
 		if (this == o) {
 			return true;
 		}
-		if (o == null || getClass() != o.getClass()) {
+		if (!(o instanceof ListView)) {
 			return false;
 		}
-		ListView<?> listView = (ListView<?>) o;
-		return Objects.equals(elementType, listView.elementType) &&
-			Objects.equals(list, listView.list);
+		final ListView<?> listView = (ListView<?>) o;
+		return getList().equals(listView.getList());
 	}
 
 	@Override
 	public int hashCode() {
-		return Objects.hash(elementType, list);
+		return Objects.hash(getList());
 	}
 
+	// --------------------------------------------------------------------------------------------
+	// Utilities
+	// --------------------------------------------------------------------------------------------
+
+	/**
+	 * Utility method for creating a {@link DataType} of {@link ListView} explicitly.
+	 */
+	public static DataType newListViewDataType(DataType elementDataType) {
+		return DataTypes.STRUCTURED(
+			ListView.class,
+			DataTypes.FIELD(
+				"list",
+				DataTypes.ARRAY(elementDataType).bridgedTo(List.class)));
+	}
+
+	// --------------------------------------------------------------------------------------------
+	// Legacy
+	// --------------------------------------------------------------------------------------------
+
+	@Deprecated
+	public transient TypeInformation<?> elementType;
```

Comment:
nit: `@Nullable`
nit: can we remove the `transient`? The whole `ListView is no longer serializable.

Reply:
No we need the transient. Otherwise it will be part of the ListView structured typed.